run-clang-format.py

```diff
@@ -0,0 +1,411 @@
+#!/usr/bin/env python
```

Comment:
I don't think this file should be part of this PR

Reply:
Oops, Sorry actually I was trying to run the clang-format test locally so I this file was created I forgot to remove it.

Reply:
Regarding `workflow_dispatch` :
By this, we can run the clang-format test manually along with other options. Actually, before submitting I was testing whether the clang-format test was running successfully or not. So I added this.
![Screenshot from 2023-02-01 19-02-30](https://user-images.githubusercontent.com/94271200/216057195-b33e406c-a4a4-4f79-a60c-f52237fff15c.png)
I was telling about the manual trigger `run workflow` button. Should I remove it??

Reply:
As long as it still runs automatically on every PR I think it's a good change yeah :) it makes it easier for contributors to try and run clang-format if they can't install clang-format-10 on their machines. And we'll eventually upgrade clang-format 14 or 15